### ISSUE TITLE
math::tests::test_tan failing (OSX)

### ISSUE THREAD

**Issue by WillBrindle** [ruffle-rs/ruffle#67]
It looks like this test is failing for me;
```
cargo test --package=ruffle_core
[...]
failures:

---- avm1::globals::math::tests::test_tan stdout ----
thread 'avm1::globals::math::tests::test_tan' panicked at 'assertion failed: `(left == right)`
  left: `Number(1.557407724654902)`,
 right: `Number(1.5574077246549023)`', core/src/avm1/globals/math.rs:216:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

The test is passing fine in circleci (https://circleci.com/gh/ruffle-rs/ruffle/309) so I assume it is something specific to my setup. Dump of obvious details below. I guess my question is whether we would expect any variation based on setup or if the value needs to be exactly consistent across all environments in order to be a faithful emulation (i.e. is it a case of loosening up the test(s) slightly to check they're approx equal to 1.0e-12 or something or is it a bug that should be addressed). 

cargo 1.37.0 (9edd08916 2019-08-02)
OS: osx 10.13.6
Processor: 2.9 GHz Intel Core i7
Memory: 16 GB 2133 MHz LPDDR3
Graphics: Radeon Pro 560 4096 MB, Intel HD Graphics 630 1536 MB

**Comment by Herschel** [comment]
Flash didn't guarantee cross-platform consistency for any of the transcendantal functions itself ([source](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/Math.html)), and I know that AS physics engines were not deterministic across platforms, so I think variations are okay and expected.

To start, we should loosen the tests using something like the [approx crate](https://docs.rs/approx/0.3.2/approx/). (This would be good to document what we precision we guarantee too, e.g. within N ULPs)

Longer term, maybe we could offer the [libm crate](https://github.com/rust-lang/libm) as an optional feature for consistency across platforms (Rust already uses this for math in wasm).

**Comment by WillBrindle** [comment]
@Herschel Sounds good. That's the crate I had in mind when I mentioned adjusting the tests. I'll take a look at doing so.

**Comment by Dinnerbone** [comment]
For these specific tests, I don't think we should aprox them. We can fix the test to reference the native constants and guarantee equality that way. Aprox may still be useful if we have hardcoded floating point math tests later, but that's not the case atm.

**Comment by WillBrindle** [comment]
@Dinnerbone correct that all these tests use fractions of PI that are constants in rust, and I think it would be better to make use of them. However, I was intending to add tests to other classes (starting with the matrix class), at least one of which also required this change (multiplying inverted matrix by identity; should come back to the original but the values are slightly off). I intentionally separated the 2 tasks to make the PR cleaner. Happy to do as one big PR to make the need clearer; or just close the PR and just use constants for now.

**Comment by Dinnerbone** [comment]
For this particular test we can pass in `f64::tan(1.0)` as that is exactly what is being called, just through the actionscript layer. We can do that with all of them. We're not so much testing what these Math functions do, just that they're routing to `f64::*`.

If you want to add tests for everything else, though, please go ahead and use approx for those - I do think it's a good idea for other tests, just not these particular ones.

**Comment by WillBrindle** [comment]
@Dinnerbone Thanks, makes sense. I'll do that for this particular issue, & roll in adding the approx crate to a separate PR. In your opinion what would be better in cases like this:
```
&[Value::Number(0.0)] => Value::Number(0.0)
```
or
```
&[Value::Number(0.0)] => Value::Number(f64::asin(0.0))
```

i.e. when the number is expected to be something trivial such as 0 just use that directly or just wrap every result up so they're all done in the same way? So rather than testing the result is a specific value it's more of a test result is the same as f64's methods.

**Comment by Dinnerbone** [comment]
I guess that in theory `f64::asin(0.0)` could be something that is approximately but not exactly `0` if your precision is wacky? To avoid this issue being reopened 3 years from now when someone tries to compile on a roman abacus, let's go with the more verbose option I guess!

**Comment by Herschel** [comment]
Merged in bd5bed0, thank you!
